project/didwhddks/README.rst

```diff
@@ -0,0 +1,142 @@
+===========================================
+Optimized Python Library for LCA Problem Solving
+===========================================
+
+Basic Information
+=================
+
+The Lowest Common Ancestor (LCA) of two nodes in a tree is the deepest node
+that is an ancestor of both. Given a rooted tree, the LCA problem involves
+efficiently answering multiple queries of the form:
+
+    What is the LCA of nodes u and v in the given tree?
+
+This problem is fundamental in graph theory and has applications in
+**computational biology**, **network routing**, **ontology trees**, and
+**hierarchical data structures**.
+
+Problems to Solve
+=================
+
+Lack of Built-in LCA Support in C++ and Python
+----------------------------------------------
+
+- C++ STL does not provide built-in support for LCA queries.
+- Python libraries such as ``networkx`` and ``scipy.sparse.csgraph`` provide
+  basic shortest path algorithms, but they are not optimized for fast repeated
+  LCA queries.
+- No specialized LCA algorithms like Binary Lifting, Euler Tour + RMQ, or
+  Tarjan's Offline LCA are implemented natively in standard libraries.
+
+Inefficiencies in Existing Graph Libraries
+------------------------------------------
+
+- Libraries like ``networkx`` allow generic shortest path computations, but
```

Comment:
It's generally hard to make an efficient Python interface for a graph library.  You need very large graph/tree for development and the data exchange interface takes a lot of time for design iteration.

It is very good to try.  Just be aware of the challenge.